### PR TITLE
libid3tag: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**
TODO:
- Merge https://github.com/conan-io/hooks/pull/508 hook first as this package is affected by the glibc issue.
- Add `-fno-finite-math-only`.

### CODE REVIEW

**Comment by RubenRBS** [comment]
Restarted, this PR was also affected by the grpc outtage

File: recipes/libid3tag/all/conanfile.py [hunk 1]
```diff
@@ -1,124 +1,140 @@
...
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
-    @property
-    def _is_msvc(self):
-        return self.settings.compiler == "Visual Studio" or (
-            self.settings.compiler == "clang" and self.settings.os == "Windows"
-        )
+    def requirements(self):
+        self.requires("zlib/1.3")
```
Comment (RubenRBS) [inline recipes/libid3tag/all/conanfile.py:51]: Add a zling range here too please! :)

File: recipes/libid3tag/all/CMakeLists.txt
Comment (uilianries) [inline recipes/libid3tag/all/CMakeLists.txt:1]: Reading the previous recipe, I see it used the MSBuild in Windows. Did you try the new https://docs.conan.io/2/reference/tools/microsoft/msbuild.html ?

File: recipes/libid3tag/all/conanfile.py [hunk 2]
```diff
@@ -1,124 +1,116 @@
...
 
     def validate(self):
-        if self._is_msvc and self.options.shared:
-            raise ConanInvalidConfiguration("libid3tag does not support shared library for MSVC")
-
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+        if self.settings.arch == "armv8" and self.options.shared:
+            raise ConanInvalidConfiguration("shared library build is not supported for armv8")
```
Comment (uilianries) [inline recipes/libid3tag/all/conanfile.py:55]:
```suggestion
```
I suggest trying it again. I can build it in Mac M1 locally, but probably you had problems when cross-building from Mac Intel to Mac M1. However, we added Mac M1 to the build matrix, so it should work now.

File: recipes/libid3tag/all/conanfile.py [hunk 3]
```diff
@@ -1,124 +1,118 @@
...
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
-    def requirements(self):
-        self.requires("zlib/1.2.11")
+    def layout(self):
+        basic_layout(self, src_folder="src")
```
Comment (uilianries) [inline recipes/libid3tag/all/conanfile.py:49]:
```suggestion
        if is_msvc(self):
            cmake_layout(self, src_folder="src")
        else:
            basic_layout(self, src_folder="src")
```

**Review by uilianries — APPROVED** [review]

**Review by toge — APPROVED** [review]

File: recipes/libid3tag/all/conanfile.py [hunk 4]
```diff
@@ -1,124 +1,121 @@
...
-        return getattr(self, "settings_build", self.settings)
+        if cross_building(self) and self.settings.arch == "armv8" and self.options.shared:
+            # https://github.com/conan-io/conan-center-index/pull/18987#issuecomment-1668243831
+            raise ConanInvalidConfiguration("shared library cross-building is not supported for armv8")
 
     def build_requirements(self):
-        if not self._is_msvc:
-            self.build_requires("gnu-config/cci.20201022")
-            if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-                self.build_requires("msys2/cci.latest")
```
Comment (SpaceIm) [inline recipes/libid3tag/all/conanfile.py:58]: This logic must not be removed because autotools require a shell.

**Pull request merged**